# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/DOMException.h
# Commit: e4783542b7c8
# Full Hash: e4783542b7c854ae4aa5cf60ac27e5e1ff19d56a
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-12-07 21:55:34
# Regressor Bug: 1588839
# File Overlap Count: 1
# Description:
#   Bug 1588839 - Part 1 - Add helper to allow structured cloning Error/Exception objects. r=bzbarsky
#   
#   In order to be able to reasonably debug error results from things like
#   JSWindowActor.sendQuery, we need to be able to clone errors across process
#   boundaries, so that they can be propagated to the caller that initiated a
# ==============================================================================

diff -r df371f905d3a -r e4783542b7c8 dom/base/DOMException.h
--- a/dom/base/DOMException.h	Sat Dec 07 17:45:53 2019 +0000
+++ b/dom/base/DOMException.h	Sat Dec 07 18:59:14 2019 +0000
@@ -137,7 +137,8 @@
 class DOMException : public Exception {
  public:
   DOMException(nsresult aRv, const nsACString& aMessage,
-               const nsACString& aName, uint16_t aCode);
+               const nsACString& aName, uint16_t aCode,
+               nsIStackFrame* aLocation = nullptr);
 
   NS_INLINE_DECL_REFCOUNTING_INHERITED(DOMException, Exception)
 